lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -452,6 +461,45 @@ prop_listRegisteredPools DBLayer {..} entries =
             ]
         assert (pools == (poolId <$> reverse entries))
 
+prop_unfetchedPoolMetadataRefs
+    :: DBLayer IO
+    -> [PoolRegistrationCertificate]
+    -> Property
+prop_unfetchedPoolMetadataRefs DBLayer{..} entries =
+    monadicIO (setup >> propA >> propB)
+  where
+    setup = do
+        run . atomically $ cleanDB
+        let entries' = (zip [SlotId ep 0 | ep <- [0..]] entries)
+        run . atomically $ mapM_ (uncurry putPoolRegistration) entries'
+        monitor $ classify (length entries > 10) "10+ entries"
+        monitor $ classify (length entries > 50) "50+ entries"
+
+    propA = do
+        refs <- run . atomically $ unfetchedPoolMetadataRefs 10
+        monitor $ counterexample $ unlines
+            [ "Read from DB (" <> show (length refs) <> "): " <> show refs
+            ]
+        assertWith "no more than known entries"
+            (length refs <= length entries)
```

Comment:
(Nitpick) Superfluous assertion given the next check? Or is this on purpose to fail fast?

Reply:
Hmm, indeed, I don't think both assertions below can be true and this one false. so it looks like redundant.

Reply:
Although, perhaps some redundancy is good for friendlier failure reasons 🤷‍♂️